src/include/catalog/abstract_catalog.h

```diff
@@ -0,0 +1,87 @@
+//===----------------------------------------------------------------------===//
+//
+//                         Peloton
+//
+// abstract_catalog.h
+//
+// Identification: src/include/catalog/abstract_catalog.h
+//
+// Copyright (c) 2015-17, Carnegie Mellon University Database Group
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include "catalog/column.h"
```

Comment:
Can you move some of these headers to the relevant implementation files? Try to avoid putting several headers in header files since that increases coupling and compilation time.